dataframely/_base_schema.py

```diff
@@ -198,11 +202,48 @@ def columns(cls) -> dict[str, Column]:
             columns[name]._name = name
         return columns
 
+    @classmethod
+    @abstractmethod
+    def polars_schema(cls) -> pl.Schema:
+        """Obtain the polars schema for this schema.
+
+        Returns:
+            A :mod:`polars` schema that mirrors the schema defined by this class.
+        """
+        pass
+
     @classmethod
     def primary_keys(cls) -> list[str]:
         """The primary key columns in this schema (possibly empty)."""
         return _primary_keys(cls.columns())
 
+    @classmethod
+    @abstractmethod
+    def validate(
```

Comment:
I had to move this here as an abstract class method because the generic argument to `dy.Dataframe` is only bound to `BaseSchema`, but I need to be able to call `.validate` on it.